File: .skills/openclaw-skills/skills/caizi333333/xygs-meeting-minutes/scripts/export_pdf.py
#!/usr/bin/env python3
"""
会议纪要 HTML → PDF 导出脚本
支持 Playwright (优先) 和 weasyprint 两种后端
用法: python export_pdf.py <input.html> <output.pdf>
"""
import sys
import os

def export_with_playwright(html_path: str, pdf_path: str) -> bool:
    """使用 Playwright Chromium 导出 PDF"""
    try:
        from playwright.sync_api import sync_playwright
        with sync_playwright() as p:
            browser = p.chromium.launch()
            page = browser.new_page()
            page.goto(f'file://{os.path.abspath(html_path)}')
            page.wait_for_load_state('networkidle')
            page.pdf(
                path=pdf_path,
                format='A4',
                print_background=True,
                margin={
                    'top': '20mm',
                    'right': '0mm',
                    'bottom': '20mm',
                    'left': '0mm'
                }
            )
            browser.close()
        return True
    except ImportError:
        return False
    except Exception as e:
        print(f"Playwright export failed: {e}", file=sys.stderr)
        return False

def export_with_weasyprint(html_path: str, pdf_path: str) -> bool:
    """使用 WeasyPrint 导出 PDF (备选方案)"""
    try:
        from weasyprint import HTML
        HTML(filename=html_path).write_pdf(pdf_path)
        return True
    except ImportError:
        return False
    except Exception as e:
        print(f"WeasyPrint export failed: {e}", file=sys.stderr)
        return False

def main():
    if len(sys.argv) < 3:
        print("Usage: python export_pdf.py <input.html> <output.pdf>")
        sys.exit(1)

    html_path = sys.argv[1]
    pdf_path = sys.argv[2]

    if not os.path.exists(html_path):
        print(f"Error: {html_path} not found", file=sys.stderr)
        sys.exit(1)

    # 尝试 Playwright (优先，效果最好)
    if export_with_playwright(html_path, pdf_path):
        size = os.path.getsize(pdf_path)
        print(f"OK [playwright] {pdf_path} ({size:,} bytes)")
        return

    # 回退到 WeasyPrint
    if export_with_weasyprint(html_path, pdf_path):
        size = os.path.getsize(pdf_path)
        print(f"OK [weasyprint] {pdf_path} ({size:,} bytes)")
        return

    print("Error: No PDF backend available. Install one of:", file=sys.stderr)
    print("  pip install playwright && playwright install chromium", file=sys.stderr)
    print("  pip install weasyprint", file=sys.stderr)
    sys.exit(1)

if __name__ == '__main__':
    main()

File: .skills/openclaw-skills/skills/dalianmao000/competitive-ops/scripts/export_pdf.js
#!/usr/bin/env node
/**
 * Export HTML reports to PDF using Playwright via npx
 * Waits for ECharts to fully render before PDF export
 */

const { chromium } = require('playwright');
const path = require('path');
const fs = require('fs');

async function exportPDF() {
    const args = process.argv.slice(2);
    let htmlPath = 'data/reports/html/index.html';
    let outputPath = null;

    for (let i = 0; i < args.length; i++) {
        if (args[i] === '-o' && args[i + 1]) {
            outputPath = args[i + 1];
            i++;
        } else if (!args[i].startsWith('-')) {
            htmlPath = args[i];
        }
    }

    const resolvedHtml = path.resolve(htmlPath);
    const htmlFile = path.basename(resolvedHtml, '.html');
    const date = new Date().toISOString().split('T')[0];
    const pdfDir = path.join(path.dirname(resolvedHtml), '..', 'pdf', date);
    const defaultOutput = path.join(pdfDir, `${htmlFile}-${date}.pdf`);
    const resolvedOutput = outputPath || defaultOutput;

    fs.mkdirSync(path.dirname(resolvedOutput), { recursive: true });

    console.log(`Exporting: ${resolvedHtml}`);
    console.log(`Output: ${resolvedOutput}`);

    const browser = await chromium.launch({ headless: true });
    const page = await browser.newPage();

    await page.setViewportSize({ width: 1400, height: 1000 });
    await page.goto(`file://${resolvedHtml}`, { waitUntil: 'networkidle' });

    // Wait for ECharts to fully render
    await page.waitForTimeout(3000);

    // Check if ECharts rendered
    const chartCount = await page.evaluate(() => {
        return document.querySelectorAll('[_echarts_instance_]').length;
    });
    console.log(`ECharts instances found: ${chartCount}`);

    await page.pdf({
        path: resolvedOutput,
        format: 'A4',
        printBackground: true,
        margin: { top: '15mm', bottom: '15mm', left: '15mm', right: '15mm' },
        displayHeaderFooter: true,
        headerTemplate: '<div></div>',
        footerTemplate: `
            <div style="width: 100%; font-size: 10px; text-align: center; color: #666;">
                Page <span class="pageNumber"></span> of <span class="totalPages"></span>
                | competitive-ops v2 | ${date}
            </div>
        `
    });

    await browser.close();

    const stats = fs.statSync(resolvedOutput);
    const sizeKb = (stats.size / 1024).toFixed(1);

    console.log(`\n✅ PDF exported successfully!`);
    console.log(`   Path: ${resolvedOutput}`);
    console.log(`   Size: ${sizeKb} KB`);
}

exportPDF().catch(err => {
    console.error(`❌ Export failed: ${err.message}`);
    process.exit(1);
});
